src/main/scala/rps/RPSMoves.scala

```diff
@@ -0,0 +1,37 @@
+package rps
+
+
+/*
+
+Scala has a strange view of Enumerations, that is
+a little bit too much for the average use-case.
+
+The "agile-version" of enumerations are called
+"sealed trait extended by case objects".
+
+Here we define a RPSMoves ADT that will contain
+useful definitions and methods to craft RPS Moves.
+
+*/
+
+
+object RPSMoves {
+
+sealed abstract class RPSMove(
+    val icon : String,
+    val value : String) {
+
+    override def toString = icon
+  }
+
+    case object ROCK extends RPSMove("💎", "0")
+    case object PAPER extends RPSMove("📄", "1")
+    case object SCISSORS extends RPSMove("✂️", "2")
+
+    // I need to find a way to obtain this set automagically
+    val setOfMoves: Set[RPSMove] = Set(ROCK, PAPER, SCISSORS)
```

Comment:
FYI, you don't need the `Set[RPSMove]` annotation

Reply:
Thank you! for clarity over brevity (in C this payed off) I usually leave all types declared, but if here in buildo we do otherwise I will change accordingly.

Do you know how to generate that Set automagically out of the enumerator?

Reply:
The general convention is that in obvious cases you leave it out. Reading the code, it's pretty obvious that this is a `Set[RPSMove]`.

In practice, you usually don't include an explicit type annotation when declaring `val`, but it's strongly encourage to always declare the result type of a public method.

> Do you know how to generate that Set automagically out of the enumerator?

Wait for the next steps ;)